cmd/frontend/backend/user_emails.go

```diff
@@ -33,56 +91,11 @@ func (userEmails) Add(ctx context.Context, userID int32, email string) error {
 	// Prevent abuse (users adding emails of other people whom they want to annoy) with the
 	// following abuse prevention checks.
 	if isSiteAdmin := CheckCurrentUserIsSiteAdmin(ctx) == nil; !isSiteAdmin {
-		if conf.EmailVerificationRequired() {
-			emails, err := db.UserEmails.ListByUser(ctx, db.UserEmailsListOptions{
-				UserID: userID,
-			})
-			if err != nil {
-				return err
-			}
-
-			var verifiedCount, unverifiedCount int
-			for _, email := range emails {
-				if email.VerifiedAt == nil {
-					unverifiedCount++
-				} else {
-					verifiedCount++
-				}
-			}
-
-			// Abuse prevention check 1: Require user to have at least one verified email address
-			// before adding another.
-			//
-			// (We need to also allow users who have zero addresses to add one, or else they could
-			// delete all emails and then get into an unrecoverable state.)
-			//
-			// TODO(sqs): prevent users from deleting their last email, when we have the true notion
-			// of a "primary" email address.)
-			if verifiedCount == 0 && len(emails) != 0 {
-				return errors.New("refusing to add additional email address for user without a verified email address")
-			}
-
-			// Abuse prevention check 2: Forbid user from having many unverified emails to prevent attackers from using this to
-			// send spam or a high volume of annoying emails.
-			const maxUnverified = 3
-			if unverifiedCount >= maxUnverified {
-				return errors.New("refusing to add email address because the user has too many existing unverified email addresses")
-			}
-		}
-		if envvar.SourcegraphDotComMode() {
-			// Abuse prevention check 3: Set a quota on Sourcegraph.com users to prevent abuse.
-			//
-			// There is no quota for on-prem instances because we assume they can trust their users
-			// to not abuse adding emails.
-			//
-			// TODO(sqs): This reuses the "invite quota", which is really just a number that counts
-			// down (not specific to invites). Generalize this to just "quota" (remove "invite" from
-			// the name).
-			if ok, err := db.Users.CheckAndDecrementInviteQuota(ctx, userID); err != nil {
-				return err
-			} else if !ok {
-				return errors.New("email address quota exceeded (contact support to increase the quota)")
-			}
+		abused, reason, err := checkEmailAbuse(ctx, userID)
+		if err != nil {
+			return err
+		} else if abused {
```

Comment:
It is normal to drop the unnecessary `else` when the `if` body returns control.
https://golang.org/doc/effective_go.html#if

```suggestion
		}
		if abused {
```

Reply:
Hmm, I feel these two should be grouped together as they both "return"s. If the `else if ...` branch does something then continues, I would totally agree with you.

Reply:
This comment is a nit and not blocking, so you are free to make a decision here.